api/provider/aws/test_aws/task_read_test.go

```diff
@@ -110,3 +110,96 @@ func TestTaskRead(t *testing.T) {
 
 	assert.Equal(t, expected, result)
 }
+
+func TestTaskRead_CannotPullContainer(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	tags := models.Tags{
+		{
+			EntityID:   "tsk_id",
+			EntityType: "task",
+			Key:        "name",
+			Value:      "tsk_name",
+		},
+		{
+			EntityID:   "tsk_id",
+			EntityType: "task",
+			Key:        "arn",
+			Value:      "arn:aws:ecs:region:012345678910:task/arn",
+		},
+		{
+			EntityID:   "tsk_id",
+			EntityType: "task",
+			Key:        "environment_id",
+			Value:      "env_id",
+		},
+		{
+			EntityID:   "dpl_id",
+			EntityType: "deploy",
+			Key:        "name",
+			Value:      "dpl_name",
+		},
+		{
+			EntityID:   "dpl_id",
+			EntityType: "deploy",
+			Key:        "version",
+			Value:      "deployVersion",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	describeTaskInput := &ecs.DescribeTasksInput{}
+	describeTaskInput.SetCluster("l0-test-env_id")
+	describeTaskInput.SetTasks([]*string{aws.String("arn:aws:ecs:region:012345678910:task/arn")})
+
+	containerECS := &ecs.Container{}
+	containerECS.SetReason("CannotPullContainerError: API error (404): repository test not found: does not exist or no pull access\n")
+	containerECS.SetExitCode(0)
+
+	task := &ecs.Task{}
+	task.SetTaskArn("arn:aws:ecs:region:012345678910:task/arn")
+	task.SetTaskDefinitionArn("arn:aws:ecs:region:account:task-definition/dpl_id:deployVersion")
+	task.SetContainers([]*ecs.Container{containerECS})
+
+	describeTaskOutput := &ecs.DescribeTasksOutput{}
+	describeTaskOutput.SetTasks([]*ecs.Task{task})
+
+	mockAWS.ECS.EXPECT().
+		DescribeTasks(describeTaskInput).
+		Return(describeTaskOutput, nil)
+
+	target := provider.NewTaskProvider(mockAWS.Client(), tagStore, mockConfig)
+	result, err := target.Read("tsk_id")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	container := models.Container{
```

Comment:
This section is asserting the entire model is correct, which is out of scope for this test. 
 For this test, we only care that the container's `ExitCode` was flipped from `0` to `1`. This whole section of of the test should be replaced with:
```go
    ...
    result, err := target.Read("tsk_id")
    if err != nil {
        t.Fatal(err)
    }

    assert.Len(t, 1, result.Containers)
    assert.Equal(t, 1, result.Containers[0].ExitCode)
}
```

Reply:
[nit] assert.Len(t, result.Containers, 1) :trollface:

Reply:
I added the length assertion so we don't hit a `panic` if we try to do ` result.Containers[0].ExitCode` when the length is 0.